File: 6-zigzag-conversion/zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if len(s) < 2 or numRows < 2:
            return s
        next = 2*numRows - 2
        st = ""
        for i in range(numRows):
            x = i
            prev = 0 
            while (x < len(s)):

                st += s[x]
                if (x+next - 2*i) != x and (x+next - 2*i) != x+next and (x+next -2*i) < len(s):
                    st += s[x+next - 2*i]
                x += next
        return st



        

File: 6-zigzag-conversion/zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if (s.length() < 2 || numRows < 2) return s;

        int next = 2 * numRows - 2;
        String st = "";

        for (int i = 0; i < numRows; i++) {
            int x = i;
            while (x < s.length()){
                st += s.charAt(x);
                if ((x+next - 2*i) != x && (x+next - 2*i) != x+next && (x+next -2*i) < s.length()){
                    st += s.charAt(x+next - 2*i);
                }
                x += next;

            }
        }
        return st;
        
    }
}